Collision/src/main/java/dk/gruppeseks/bodtrd/collision/CollisionProcessor.java

```diff
@@ -51,59 +52,87 @@ public void process(World world)
+                    handleCollision(handled, ent, entities);
+                }
+            }
+        }
+    }
 
-                    for (Entity ent2 : entities) // Checks if it collides with anything.
-                    {
-                        CollisionData ent2Data = ent.get(CollisionData.class);
-
-                        if (ent2.getID() == handled.getID() || ent2.getID() == ent.getID() || ent2Data == null)
-                        {
-                            continue;
-                        }
-                        if (CollisionHandler.isColliding(handled, ent2))
-                        {
-                            for (Entity ent3 : entities)
-                            {
-                                CollisionData ent3Data = ent.get(CollisionData.class);
-
-                                if (ent3.getID() == handled.getID() || ent3.getID() == ent.getID()
-                                        || ent3.getID() == ent2.getID() || ent3Data == null)
-                                {
-                                    continue;
-                                }
-                                if (CollisionHandler.isColliding(handled, ent3))
-                                {
-                                    handledPos.setX(startX);
-                                    handledPos.setY(startY); // If it even collides after a third collision correction, then dont do any corrections at all.
-                                    collision = true;
-                                    break;
-                                }
-
-                            }
-                            if (!collision)
-                            {
-                                // if it doesnt collide with a third object after the second collision correction, put it to the secondcalculated position.
-                                Position response = CollisionHandler.collisionResponse(handled, ent2);
-                                handledPos.setX(response.getX());
-                                handledPos.setY(response.getY());
-                            }
-                            collision = true;
-                            break;
-                        }
+    private void handleCollision(Entity handled, Entity ent, Collection<Entity> entities)
+    {
+        switch (handled.getType())
+        {
+            case PLAYER:
+            {
+                if (ent.getType() == EntityType.PROJECTILE)
+                {
 
+                }
+                else if (ent.getType() == EntityType.WALL)
+                {
+                    getBounceResponse(handled, ent, entities);
+                }
+                break;
+            }
+            case PROJECTILE:
+            {
+                if (ent.getType() == EntityType.PLAYER)
+                {
+                    Owner o = handled.get(Owner.class);
+                    ent.get(Health.class).addDamageInstance(new DamageInstance(10, o.getId()));
```

Comment:
i know health stuff isnt implemented. but this would result in a null pointer exception